File: medplum-java/medplum-model/src/main/java/com/fhir/PlanDefinition_RelatedAction.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutablePlanDefinition_RelatedAction.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface PlanDefinition_RelatedAction {
  Optional<Duration> offsetDuration();

  Optional<id> actionId();

  Optional<Plandefinition_relatedactionRelationship> relationship();

  Optional<String> id();

  Optional<List<Extension>> modifierExtension();

  Optional<Range> offsetRange();

  Optional<List<Extension>> extension();

  static ImmutablePlanDefinition_RelatedAction.Builder builder() {
    return ImmutablePlanDefinition_RelatedAction.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/PlanDefinition_RelatedAction.ts
import { Duration } from "./Duration";
import { Extension } from "./Extension";
import { id } from "./id";
import { Plandefinition_relatedactionRelationship } from "./Plandefinition_relatedactionRelationship";
import { Range } from "./Range";

export interface PlanDefinition_RelatedAction {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    actionId: id | null | undefined;
    relationship: Plandefinition_relatedactionRelationship | null | undefined;
    offsetDuration: Duration | null | undefined;
    offsetRange: Range | null | undefined;
}
